plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/DBUtils.java

```diff
@@ -1390,10 +1391,17 @@ public static DBCStatement makeStatement(
             makeStatement(session, queryText, doScrollable);
         dbStat.setStatementSource(executionSource);
 
+        boolean isShouldSetLimit = true;
```

Comment:
Please move it to line 1351 or around. Also leave transformProvider var declaration where it was before. It is not needed in other code blocks.

Reply:
I can't have transformProvider var declaration inside that if block as I need transformProvider here on the line 1395.